llvm/lib/Target/AMDGPU/AMDGPU.h

```diff
@@ -130,6 +130,11 @@ struct AMDGPULowerKernelAttributesPass
   PreservedAnalyses run(Function &F, FunctionAnalysisManager &AM);
 };
 
+struct AMDGPUOclcReflectPass : public PassInfoMixin<AMDGPUOclcReflectPass> {
+public:
+  PreservedAnalyses run(Function &M, FunctionAnalysisManager &AM);
```

Comment:
One thing to note here is that (I believe) this pass won't run at `-O0` and in other configurations because it's not marked as `static bool isRequired() { return true; }`.

Perhaps that's what we want, because it's currently essentially an "optimization", but we should probably document it (somewhere). Someone might want to use it as a requirement (do this new reflection otherwise the backend fails due to a bug) and then we'd have to reconsider this choice.

Reply:
This is a good catch, thanks. I think we always want this to run regardless of the opt level. The alternative is that we get orphaned calls to `__oclc_amdgpu_reflect` that cause a linker error if the user uses some math func that relies on AMDGPU reflecting

Reply:
Have changed. This is the only pass using this check in AMDGPU but it is indeed being checked